### DISCUSSION TITLE
ArcadeDB Studio

### DISCUSSION THREAD

**Discussion by lvca** [ArcadeData/arcadedb#23] [Announcements]
Hi guys,

After @grimace opened an issue about the [lack of a home page when you run ArcadeDB](https://github.com/ArcadeData/arcadedb/issues/18), I quickly wrote a super basic studio (inspired to OrientDB Studio)

After authenticating you can execute commands:

![image](https://user-images.githubusercontent.com/312606/131901586-948b5634-7d3c-439b-8df7-b32c0ef0e2ec.png)

Check the available HTTP API:

![image](https://user-images.githubusercontent.com/312606/131901832-58126ee8-dead-486d-8758-776002bd3d7d.png)

And have some useful resources:

![image](https://user-images.githubusercontent.com/312606/131901869-c28158f8-954e-4a5f-8c34-9760dac70214.png)

The studio is super basic, with just a bit of jquery, bootstrap, and ajax calls. If you'd like to create something fancier as an external plugin, you're more than welcome to do that. Please let us know how can we help.

You can find the docs at: https://docs.arcadedb.com/#Studio

**Comment by lvca** [comment]
## SUMMARY

Status: Completed
Progress: Design 100% -> Development & Unit Tests 100% -> Integration Tests 100%
Currently Assigned: @lvca 
ETA: 6 days

---

I worked a lot on Studio in the past week adding a lot of new features, like:
- graph support
- create/drop/import databases
- authentication
- improved table display
- lazy loading of neighbors in the graph
- shorted path algorithm
- editing of elements in the graph
- code highlighting for SQL,MongoDB and Cypher

**Comment by Praveenstein** [comment]
Hi, I'm new to this database (and graph database too), can you please tell me how to use the studio and gremlin to create a new vertex, we usually use g.V().addV("person") etc, but we need to define g, how do i do it in the studio, and is there any cookbook or recipe blogs to help beginners with doing the basic CRUD operations (in all supported languages)?? I couldn't find any such recipes in the official documentation. Thanks in advance

**Reply by lvca** [reply]
This is a good idea, one cookbook per supported language. I'll add to the docs project: https://github.com/ArcadeData/arcadedb-docs/issues/86

**Reply by gramian** [reply]
Indeed, there are no recipes in the documentation for other languages than SQL, but there are links to pages about these query languages at the end of the respective sections, see:

* Cypher: https://docs.arcadedb.com/#Open-Cypher
* Gremlin: https://docs.arcadedb.com/#Gremlin-API

I know this does not resolve the general problem, but it may help resolve the immediate pain. :)

**Reply by Praveenstein** [reply]
Thanks, I'm also working on creating my own manual (for graph database in general, gremlin language, and how to use all these with arcade DB), hopefully, if I could complete it, I'll make it public

**Reply by lvca** [reply]
@Praveenstein feel free to share a draft if you can :-)
